### PR TITLE
fix(clerk-react): Proper error on missing API keys

### PR DESCRIPTION
**Pull request by agis**
## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [x] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `@clerk/shared`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

Before this change, when neither of the keys were set (publishableKey or frontendApi), the following error would be thrown:

    The publishableKey passed to Clerk is invalid. You can get your
    Publishable key at
    https://dashboard.clerk.dev/last-active?path=api-keys. (key=)

This was misleading, since it implies that publishableKey is the only actionable from the user, whereas someone has to set either publishableKey _or_ frontendApi. Therefore, this patch changes the error message to the following when both keys are missing:

    Missing frontendApi or publishableKey. You can get your key at
    https://dashboard.clerk.dev/last-active?path=api-keys.

Similarly it renders more relevant messages when either of the keys were passed but was invalid.

Fixes AUTH-121

### CODE REVIEW

File: packages/react/src/contexts/ClerkProvider.tsx [hunk 1]
```diff
@@ -21,8 +21,12 @@ function ClerkProviderBase(props: ClerkProviderProps): JSX.Element {
   const { initialState, children, ...restIsomorphicClerkOptions } = props;
   const { frontendApi = '', publishableKey = '' } = restIsomorphicClerkOptions;
 
-  if (!isPublishableKey(publishableKey) && !isLegacyFrontendApiKey(frontendApi)) {
-    errorThrower.throwInvalidPublishableKeyError({ key: publishableKey || frontendApi || '' });
+  if (publishableKey == '' && frontendApi == '') {
+    errorThrower.throwMissingFrontendApiPublishableKeyError();
+  } else if (publishableKey != '' && !isPublishableKey(publishableKey)) {
+    errorThrower.throwInvalidPublishableKeyError({ key: publishableKey });
+  } else if (frontendApi != '' && !isLegacyFrontendApiKey(frontendApi)) {
```
Comment (chanioxaris) [inline packages/react/src/contexts/ClerkProvider.tsx:28]:
```suggestion
  if (!publishableKey && !frontendApi) {
    errorThrower.throwMissingFrontendApiPublishableKeyError();
  } else if (publishableKey && !isPublishableKey(publishableKey)) {
    errorThrower.throwInvalidPublishableKeyError({ key: publishableKey });
  } else if (frontendApi && !isLegacyFrontendApiKey(frontendApi)) {
```

**Review by chanioxaris — APPROVED** [review]
Awesome

**Review by SokratisVidros — APPROVED** [review]

**Review by nikosdouvlis — APPROVED** [review]

File: packages/react/src/contexts/ClerkProvider.tsx [hunk 2]
```diff
@@ -21,8 +21,12 @@ function ClerkProviderBase(props: ClerkProviderProps): JSX.Element {
   const { initialState, children, ...restIsomorphicClerkOptions } = props;
   const { frontendApi = '', publishableKey = '' } = restIsomorphicClerkOptions;
 
-  if (!isPublishableKey(publishableKey) && !isLegacyFrontendApiKey(frontendApi)) {
-    errorThrower.throwInvalidPublishableKeyError({ key: publishableKey || frontendApi || '' });
+  if (!publishableKey && !frontendApi) {
+    errorThrower.throwMissingFrontendApiPublishableKeyError();
```
[the remark this answers is not in the document]
Comment (agis) [reply packages/react/src/contexts/ClerkProvider.tsx]: done

[the remark this answers is not in the document]
Comment (SokratisVidros) [reply packages/react/src/contexts/ClerkProvider.tsx]: All set :)

**Pull request merged**